src/main/host/thread_ptrace.c

```diff
@@ -1079,41 +1079,63 @@ bool threadptrace_isRunning(Thread* base) {
     return false;
 }
 
-void threadptrace_terminate(Thread* base) {
-    ThreadPtrace* thread = _threadToThreadPtrace(base);
-
-    /* make sure we cleanup circular refs */
-    if (thread->sys) {
-        syscallhandler_unref(thread->sys);
-        thread->sys = NULL;
-    }
-
-    if (!threadptrace_isRunning(base)) {
+static void _threadptrace_terminateHelper(ThreadPtrace* thread) {
```

Comment:
Optional nit - I'm not sure this helper is buying us much since it's only called from one place, and the caller doesn't do much besides call it. The decision about what to put here rather in the caller seems a bit arbitrary (though maybe a more precise name for this helper would help)